python/cuml/tsa/arima.pyx

```diff
@@ -0,0 +1,817 @@
+    best_mu = np.zeros(num_batches)
+    best_ar_params = num_batches*[None]
+    best_ma_params = num_batches*[None]
+
+    for p in range(0, max_p):
+        arparams = np.zeros(p)
+        for q in range(0, max_q):
+            maparams = np.zeros(q)
+
+            # skip 0,0 case
+            if p == 0 and q == 0:
+                continue
+
+            mu0, ar0, ma0 = estimate_x0((p, d, q), 2, y_b)
+
+            b_model = fit(y_b, (p, d, q), mu0, ar0, ma0)
+
+            if method == "aic":
+                ic = b_model.aic
+            elif method == "bic":
+                ic = b_model.bic
+            else:
+                raise NotImplementedError("Method '{}' not supported".format(method))
+
+            for (i, ic_i) in enumerate(ic):
+                if ic_i < best_ic[i]:
+                    best_order[i] = (p, d, q)
+                    best_mu[i] = b_model.mu[i]
+
+                    if p > 0:
+                        best_ar_params[i] = b_model.ar_params[i]
+                    else:
+                        best_ar_params[i] = []
+                    if q > 0:
+                        best_ma_params[i] = b_model.ma_params[i]
+                    else:
+                        best_ma_params[i] = []
+
+                    best_ic[i] = ic_i
+
+    return (best_order, best_mu, best_ar_params, best_ma_params, best_ic)
+
+
+def unpack(p, d, q, nb, np.ndarray[double, ndim=1] x):
+    """Unpack linearized parameters into mu, ar, and ma batched-groupings"""
+    nvtx_range_push("unpack(x) -> (ar,ma,mu)")
+    num_parameters = d + p + q
+    mu = np.zeros(nb)
+    ar = []
+    ma = []
+    for i in range(nb):
+        xi = x[i*num_parameters:(i+1)*num_parameters]
+        if d > 0:
+            mu[i] = xi[0]
+        if p > 0:
+            ar.append(xi[d:(d+p)])
+        ma.append(xi[d+p:])
+
+    nvtx_range_pop()
+    return (mu, ar, ma)
+
+
+def pack(p, d, q, nb, mu, ar, ma):
+    """Pack mu, ar, and ma batched-groupings into a linearized vector `x`"""
+    nvtx_range_push("pack(ar,ma,mu) -> x")
+    num_parameters = d + p + q
+    x = np.zeros(num_parameters*nb)
+    for i in range(nb):
+        xi = np.zeros(num_parameters)
+        if d > 0:
+            xi[0] = mu[i]
+
+        for j in range(p):
+            xi[j+d] = ar[i][j]
+        for j in range(q):
+            xi[j+p+d] = ma[i][j]
+        # xi = np.array([mu[i]])
+        # xi = np.r_[xi, ar[i]]
+        # xi = np.r_[xi, ma[i]]
+        x[i*num_parameters:(i+1)*num_parameters] = xi
+
+    nvtx_range_pop()
+    return x
+
+
+def _batched_transform(p, d, q, nb, np.ndarray[double] x, isInv, handle=None):
+    cdef vector[double] vec_ar
+    cdef vector[double] vec_ma
+    cdef vector[double] vec_Tar
+    cdef vector[double] vec_Tma
+
+    nvtx_range_push("batched_transform")
+
+    if handle is None:
+        handle = cuml.common.handle.Handle()
+    cdef cumlHandle* handle_ = <cumlHandle*><size_t>handle.getHandle()
+    cdef np.ndarray[double] Tx = np.zeros(nb*(d+p+q))
```

Comment:
Not sure if this would still trigger an ABI incompatibility issue we used to have with numpy from the deafault conda channel. This can be changed to 
```suggestion
    Tx = np.zeros(nb*(d+p+q), dtype=np.float64)
```

And then the pointer can be easily extracted with `Tx.ctypes.data` instead of `&Tx[0]` (though I'm not entirely certain that the latter won't still work even with the change of the definition of `Tx`

Reply:
Maybe i'm not understanding, but I cannot figure out how to extract the pointer from `Tx.ctypes.data`. I just get conversion errors like the following:
`cuml/tsa/arima.pyx:682:76: Cannot convert Python object to 'double *'`

Reply:
Ah forgot to mention, `Tx.ctypes.data` is the pointer, you can see the use here: https://github.com/rapidsai/cuml/blob/f77f8ef7ec0986bfaf9e1ba19c8230947ae0e23b/python/cuml/ensemble/randomforestclassifier.pyx#L679 and then sent to C++ in https://github.com/rapidsai/cuml/blob/f77f8ef7ec0986bfaf9e1ba19c8230947ae0e23b/python/cuml/ensemble/randomforestclassifier.pyx#L704

Reply:
Thanks for the reference. I fixed it to remove the `cimport numpy as np`.